test/factory/js_factory_test.js

```diff
@@ -7,3 +7,13 @@ window._JsFoo = class JsFooComponent extends React.Component {
 window._JsFooFunction = React.forwardRef((props, ref) => (
   React.createElement("div", {...props, ref: ref})
 ));
+
+window._JsNoChildren = class JsNoChildrenComponent extends React.Component {
+  render() {
+    if(this.props.children !== undefined) {
+      throw Error('children prop must be undefined');
+    }
```

Comment:
I don't know if there's a more straightforward way to test that `props.children` is specifically `undefined` and not `null`

Reply:
I thought maybe we could change [this expectation](https://github.com/Workiva/react-dart/blob/master/test/factory/common_factory_tests.dart#L55):
```diff
-expect(getChildren(instance), shouldAlwaysBeList ? [] : isNull);
+// We want to make sure this is undefined and not null, so we have to use `same`
+// which uses `identical`, since in Dart null and undefined are otherwise interchangeable.
+expect(getChildren(instance), shouldAlwaysBeList ? [] : same(jsUndefined));
```

But it looks like that doesn't even work, since the `identical` check gets compiled to `==` instead of `===` , and `identical(jsNull, jsUndefined)` is `true` 😕.

So, checking it from the JS seems like the way to go to me!

Related to my comment above, we could also potentially check whether it's undefined on the ReactElement without rendering it, if the rendering/throwing causes issues or complicates things.
```tsx
window.hasUndefinedChildren = (reactElement) => reactElement.props.children === undefined;
```
```dart
 expect(hasUndefinedChildren(JsNoChildren({})), isTrue);
```

Reply:
Oh that's a lot nicer! Thank you!